TrainingTests/RepositoryViewModelTests.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  RepositoryViewModelTests.swift
+//  TrainingTests
+//
+//  Created by Ong Yue Huei on 19/07/2021.
+//
+
+import Combine
+@testable import Training
+import XCTest
+
+class RepositoryViewModelTests: XCTestCase {
+    private var getRepoUseCase = GetRepoUseCaseMock()
+    private lazy var dependency = RepositoryViewModel.Dependency(getRepoUseCase: getRepoUseCase)
+   
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+    }
+
+    override func tearDownWithError() throws {
+        try super.tearDownWithError()
+    }
+
+    func test_init() {
+        let vm = createViewModel()
+        XCTAssertTrue(getRepoUseCase.callArgs.isEmpty)
+        XCTAssertNil(vm.state.repo)
+    }
+    
+    func test_fetch() {
+        let vm = createViewModel()
+        let repoResponse = Repo.stub()
+        getRepoUseCase.publisher = Result.Publisher(repoResponse).eraseToAnyPublisher()
+
+        vm.send(action: .fetch)
+        XCTAssertEqual(getRepoUseCase.callArgs, [.perform(ownerRepo: Event.stub()[0].repo.name)])
+        XCTAssertEqual(vm.state.repo, repoResponse)
+    }
```

Comment:
`stub()`はランダムな値として扱って欲しいので、expect valueとして使わないようにして欲しいです！

`createViewModel(argument:)`で`event`を設定して、その`event`で`getRepoUseCase.perform(owerRepo:)`が call されるのをテストするようにしてほしいです！

```suggestion
    func test_fetch() {
        let event = Event.stub()
        let vm = createViewModel(argument: .init(event: event))
        let repoResponse = Repo.stub()
        getRepoUseCase.publisher = Result.Publisher(repoResponse).eraseToAnyPublisher()

        vm.send(action: .fetch)
        XCTAssertEqual(getRepoUseCase.callArgs, [.perform(ownerRepo: event.repo.name)])
        XCTAssertEqual(vm.state.repo, repoResponse)
    }
```

Reply:
承知しました！ありがとうございます！